tests/test_wave_enhancement.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+
+def test_newest_by_category():
+    item_a = Clothing(condition=2.0, age=8.0)
+    item_b = Decor(condition=2.0, age=8.0)
+    item_c = Clothing(condition=4.0, age=3.0)
+    item_d = Decor(condition=5.0, age=1.0)
+    item_e = Clothing(condition=3.0, age=5.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    newest_item = tai.get_newest_by_category("Clothing")
+
+    assert newest_item.category == "Clothing"
+    assert newest_item.age == pytest.approx(3.0)
+
+
+def test_best_by_category_no_matches_is_none():
+    item_a = Decor(condition=2.0, age=8.0)
+    item_b = Decor(condition=2.0, age=8.0)
+    item_c = Decor(condition=4.0, age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    newest_item = tai.get_best_by_category("Electronics")
+
+    assert newest_item is None
+
+
+def test_newest_by_category_with_duplicates():
+    item_a = Clothing(condition=2.0, age=8.0)
+    item_b = Clothing(condition=4.0, age=3.0)
+    item_c = Clothing(condition=4.0, age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    newest_item = tai.get_newest_by_category("Clothing")
+
+    assert newest_item.category == "Clothing"
+    assert newest_item.age == pytest.approx(3.0)
+
+
+def test_swap_newest():
+    item_a = Decor(condition=2.0, age=8.0)
+    item_b = Electronics(condition=4.0, age=3.0)
+    item_c = Decor(condition=4.0, age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    item_d = Clothing(condition=2.0, age=8.0)
+    item_e = Decor(condition=4.0, age=3.0)
+    item_f = Clothing(condition=4.0, age=3.0)
+    jesse = Vendor(
+        inventory=[item_d, item_e, item_f]
+    )
+
+    result = tai.swap_by_newest(
+        other=jesse,
+        my_priority="Clothing",
+        their_priority="Decor"
+    )
+
+    assert result is True
+    assert len(tai.inventory) is 3
+    assert len(jesse.inventory) is 3
+    assert item_c not in tai.inventory
+    assert item_f in tai.inventory
+    assert item_f not in jesse.inventory
+    assert item_c in jesse.inventory
+
+
+def test_swap_by_newest_no_match_is_false():
```

Comment:
Good pair of edge-case tests.

You should also test when there are items tied for newest.